### PR TITLE
Gpl409 integrate lighthouse service

### LINKED ISSUE #130: GPL-409 Create plate in SS programatically by calling the lighthouse service
**User story**
As a dev team we would like AA to call the lighthouse service when the 'receive_plate' process occurs (similar to tube racks) so that the lighthouse service can find the sample data and create a plate in SS with that data.

**Who the primary contacts are for this story**
Philip

**Acceptance criteria**
To be considered successful the solution must allow:
1. the lighthouse service needs to be called (HTTP POST) with each barcode scanned in with the 'receive_plate' process

**Additional context**
Blocked by sanger/crawler#2

### PR DESCRIPTION
**Pull request by pjvv**
Closes #130 

Changes proposed in this pull request:

* call lighthouse service when scanning in plates

### CODE REVIEW

File: app/controllers/process_plates_controller.rb [hunk 1]
```diff
@@ -41,4 +55,21 @@ def create
...
+
+  private
+
+  # Returns a list of unique barcodes by removing blanks and duplicates
+  def sanitize_barcodes(barcodes)
+    sanitized_barcodes = [barcodes].map do |s|
+      s.split(/\s/).reject(&:blank?)
+    end
+    sanitized_barcodes.flatten.uniq
+  end
```
Comment (emrojo) [inline app/controllers/process_plates_controller.rb:67]: Other option could be:
barcodes.split(/\s+/).reject(&:blank?).uniq

File: app/models/lighthouse.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    end
+
+    Rails.logger.info("Sent POST requests to lighthouse service: #{responses}")
+
+    responses
+  rescue StandardError => e
+    Rails.logger.error(e)
+    nil
+  end
+end
```
Comment (emrojo) [inline app/models/lighthouse.rb:32]: I would move this file to the /lib folder, as it is a library.

File: app/controllers/process_plates_controller.rb (hunk 1 above)
Comment (pjvv) [reply app/controllers/process_plates_controller.rb:64]: Thanks, I wanted to ask how to test this private method in the controller too.

Comment (emrojo) [reply app/controllers/process_plates_controller.rb:64]: We cannot access private methods in tests to my knowledge. If we want to test it specifically we could move it to the helpers module or make it public. 
If is not very important this function (not being called in other places), the test from the public method can test this functionality.

**Pull request merged**